### PR TITLE
ManagedLedger should share BK worker pool threads

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

Currently, both BookKeeper client and ManagedLedger have their own worker thread pools. 
There is no advantage in that and it just lead to having many "critical" threads and more context switches.

### CODE REVIEW

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerFactoryImpl.java [hunk 1]
```diff
@@ -126,10 +127,18 @@ private ManagedLedgerFactoryImpl(BookKeeper bookKeeper, boolean isBookkeeperMana
                 .numThreads(config.getNumManagedLedgerSchedulerThreads())
                 .name("bookkeeper-ml-scheduler")
                 .build();
-        orderedExecutor = OrderedExecutor.newBuilder()
-                .numThreads(config.getNumManagedLedgerWorkerThreads())
-                .name("bookkeeper-ml-workers")
-                .build();
+
+        if (bookKeeper.getMainWorkerPool() != null) {
```
Comment (ivankelly) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerFactoryImpl.java]: This is a public interface, but not a good one. And ownership is all messed up.

Does this even give a real performance benefit? The managed ledger selects the thread based on ml name, the ledger by the ledger id. There's no guarantee these will be the same. A better approach would be to allow the managed ledger to pass in its executor to the ledger on creation.

**Comment by codelipenghui** [comment]
@merlimat Please take a look at the comment. Is this PR need to onboard the 2.6.0 release?

**Comment by codelipenghui** [comment]
move to 2.7.0

**Comment by codelipenghui** [comment]
move to 2.8.0

**Pull request closed without merging**